src/mono/mono/mini/interp/transform.c

```diff
@@ -1479,6 +1479,37 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 	} else if (in_corlib && !strcmp (klass_name_space, "System") && !strcmp (klass_name, "ByReference`1")) {
 		g_assert (!strcmp (tm, "get_Value"));
 		*op = MINT_INTRINS_BYREFERENCE_GET_VALUE;
+	} else if (in_corlib && !strcmp (klass_name_space, "System") && !strcmp (klass_name, "Marvin")) {
+		if (!strcmp (tm, "Block"))
+			*op = MINT_INTRINS_MARVIN_BLOCK;
```

Comment:
I'd just intrinsify `String.GetHashCode` and implement it using the non-random algorithm like this: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs#L781-L783 should be way faster and that type of attack usually back-end only

Reply:
Doesn't sound like a good idea to have the interpreter just use a different hash function compared to jit/aot or coreclr. Arguably, maybe I should intrinsify both `Marvin.ComputeHash32` and  `Marvin.ComputeHash32OrdinalIgnoreCase`.

Reply:
I'd suggest to use this one for all Xamarin workloads and all modes including AOT.

also it doesn't matter if two different implementations return different hash codes - string.GetHashCode already returns different hash codes for the same input between sessions - it uses a random seed.

Reply:
@EgorBo I think we should discuss that in a separate issue.